src/NuGet.Indexing/Extraction/DocumentCreator.cs

```diff
@@ -261,6 +263,56 @@ private static void AddRequiresLicenseAcceptance(Document document, IDictionary<
             }
         }
 
+        private static void AddPackageTypes(Document document, IDictionary<string, string> package)
+        {
+            string value;
```

Comment:
maybe rename this to `flattenedPackageTypes` so it's more clear what it is at a glance?